src/terminal/parser/stateMachine.cpp

```diff
@@ -1270,6 +1437,233 @@ void StateMachine::_EventVt52Param(const wchar_t wch)
     }
 }
 
+// Routine Description:
+// - Processes a character event into an Action that occurs while in the DcsEntry state.
+//   Events in this state will:
+//   1. Execute C0 control characters
+//   2. Ignore Delete characters
+//   3. Store parameter data
+//   4. Collect Intermediate characters
+//   5. Pass through characters
+//   6. If we see a ESC, enter the DcsTermination state.
+//  DCS sequences are structurally almost the same as CSI sequences, just with a
+//      extra data string. It's safe to reuse CSI functions for
+//      determining if a character is a parameter, delimiter, or invalid.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsEntry(const wchar_t wch)
+{
+    _trace.TraceOnEvent(L"DcsEntry");
+    if (_isC0Code(wch))
+    {
+        _ActionExecute(wch);
+    }
+    else if (_isDelete(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isCsiInvalid(wch))
+    {
+        _EnterDcsIgnore();
+    }
+    else if (_isNumericParamValue(wch) || _isCsiDelimiter(wch))
+    {
+        _ActionParam(wch);
+        _EnterDcsParam();
+    }
+    else if (_isIntermediate(wch))
+    {
+        _ActionCollect(wch);
+        _EnterDcsIntermediate();
+    }
+    else if (_isDcsPassThroughValid(wch))
+    {
+        _ActionDcsPassThrough(wch);
+        _EnterDcsPassThrough();
+    }
+    else if (_isDcsTerminationInitiator(wch))
+    {
+        _EnterDcsTermination();
+    }

```

Comment:
I don't think we should be entering the termination state from anything other than the pass through state. Until then we don't have a valid sequence, so an escape should just be aborting the partial DCS and attempting to process the new escape (and that's handled automatically in the `ProcessCharacter` method).

Reply:
Yeah you're right. Aside from pass through state, iTerm2 also allows entering the termination state from `DcsIgnore` using `ESC\`.